### PR TITLE
Allow converters to delegate to others.

### PR DESCRIPTION
**Pull request by JakeWharton**
Add a sample for this in the form of one that forces chunked uploads when `@Chunked` is added to `@Body`.

Refs #1315.

### CODE REVIEW

File: retrofit/src/main/java/retrofit/Retrofit.java [hunk 1]
```diff
@@ -225,13 +225,25 @@ public BaseUrl baseUrl() {
...
+   *
+   * @throws IllegalArgumentException if no converter available for {@code type}.
+   */
+  public <T> Converter<T, RequestBody> nextRequestBodyConverter(Converter.Factory skipPast,
+      Type type, Annotation[] annotations) {
     checkNotNull(type, "type == null");
     checkNotNull(annotations, "annotations == null");
 
-    for (int i = 0, count = converterFactories.size(); i < count; i++) {
+    int start = converterFactories.indexOf(skipPast) + 1;
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit/Retrofit.java]: This Just Works with -1!

File: samples/src/main/java/com/example/retrofit/ChunkingConverter.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import retrofit.Converter;
+import retrofit.GsonConverterFactory;
+import retrofit.Retrofit;
+import retrofit.http.Body;
+import retrofit.http.POST;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+public final class ChunkingConverter {
```
Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ChunkingConverter.java]: This is a craxy example.

Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ChunkingConverter.java]: (I think it's a bit distracting; normals shouldn't need to do this)

Comment (JakeWharton) [inline samples/src/main/java/com/example/retrofit/ChunkingConverter.java]: Yeah I wrote it for myself to prove the implementation and figured I'd include it in the PR for kicks. Happy to remove, but any idea for other examples?

Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ChunkingConverter.java]: Luhny-bin? Rewrite a text file to replace all ascii characters and digits with X ?

Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ChunkingConverter.java]: (that'd be a String to String conversion)

**Pull request merged**